app/services/push_notification.rb

```diff
@@ -14,16 +14,7 @@ def self.send_notifications(message_options = {})
     @client = Exponent::Push::Client.new
     Notification::Device.all.each do |device|
       device_token = device.token
-
-      # Send PushNotification
-      messages = [message_options.merge(to: device_token)]
-
-      begin
-        feedback = @client.send_messages(messages)
-        RedisAdapter.add_push_log(device_token, message_options.merge(date: DateTime.now, payload: feedback.try(:response).try(:body)))
-      rescue => e
-        RedisAdapter.add_push_log(device_token, message_options.merge(rescue_error: "push notification", error: e, date: DateTime.now))
-      end
+      SendSinglePushNotificationJob.perform_later(device_token, message_options)
```

Comment:
dann könntste auch noch zeile 16 einsparen und `device` weiterreichen, um wie `client` und `messages` dann den `device_token` innerhalb der perform-methode zu setzen?

Reply:
Man sollte aber in delayed_job keine Objekte rein reichen. Am besten nur Springs, int4egers oder mini Hash. Alle Attribute werden ja serializiert in der Datenbank gespeichert.